earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -101,66 +125,195 @@ void setDefaultValues()
    initialDOMHeapAllocSize = 0x4000;
    maxDOMHeapAllocSize     = 0x20000;
    maxDOMSubAllocationSize = 0x1000;
+   terminateCache = false;
+   percent = 1.5;
+}
+
+void getInitValues()
+{
+    static bool callGuard = false;
+    if (callGuard) return;
+    callGuard = true;
+    setDefaultValues();
+    try
+    {
+        for(const auto& i : options)
+        {
+            config_parser.addOption(i);
+        }
+        config_parser.parse(XMLConfigFile.c_str());
+        config_parser.validateIntegerValues();
+        for (const auto& it : config_parser)
+        {
+            if (it.first == INIT_HEAP_SIZE)
+                initialDOMHeapAllocSize = std::stol(it.second);
+            else if (it.first == MAX_HEAP_SIZE)
+                maxDOMHeapAllocSize = std::stol(it.second);
+            else if (it.first == BLOCK_SIZE)
+                maxDOMSubAllocationSize = std::stol(it.second);
+            else if (it.first == PURGE)
+                terminateCache = std::stoi(it.second);
+            else
```

Comment:
I agree with @tst-jhaight here - we should check explicitly for PURGE_LEVEL and then have an else that prints out an error for invalid options.